src/zip.c

```diff
@@ -197,6 +233,15 @@ struct zip_t *zip_open(const char *zipname, int level, char mode) {
       goto cleanup;
     }
     break;
+  case 'd':
+    if (zip_reader_writer_init_file(
+            &(zip->archive), zipname,
+            zip->level | MZ_ZIP_FLAG_DO_NOT_SORT_CENTRAL_DIRECTORY)) {
+      // An archive file does not exist or cannot initialize
+      // zip_archive reader
+      goto cleanup;
+    }
+    break;
```

Comment:
Yeah, we can simplify the code here!  Since the value of the variable `pZip->m_central_directory_file_ofs` is reset in function `mz_zip_writer_init_from_reader()`, we just need to replace `m_central_directory_file_ofs` with `m_archive_size` in the subsequent code.